Update dependencies

tamird opened the pull request:
Not very much to see here - the work is mostly in nailing down various constraints in `Gopkg.toml` which were not previously defined or documented. That said, it's late in the cycle, so I'm fine with waiting until after 1.1 to merge this, but as far as I can tell, there's nothing particularly interesting here.

Notable changes:
- golang.org/x/crypto/ssh: require host key checking in the ClientConfig (https://github.com/golang/crypto/commit/e4e2799)
- golang.org/x/crypto/ssh/knownhosts: a parser for the OpenSSH known_hosts file format (https://github.com/golang/crypto/commit/ed779e1)
- github.com/golang/lint: Ignore generated code (https://github.com/golang/lint/commit/a5f4a24)
- google.golang.org/grpc: Do not create new addrConn when connection error happens (https://github.com/grpc/grpc-go/commit/5ad7775)

tamird committed:
Update dependencies

Notable changes:
- golang.org/x/crypto/ssh: require host key checking in the ClientConfig (https://github.com/golang/crypto/commit/e4e2799)
- golang.org/x/crypto/ssh/knownhosts: a parser for the OpenSSH known_hosts file format (https://github.com/golang/crypto/commit/ed779e1)
- github.com/golang/lint: Ignore generated code (https://github.com/golang/lint/commit/a5f4a24)
- google.golang.org/grpc: Do not create new addrConn when connection error happens (https://github.com/grpc/grpc-go/commit/5ad7775)
- google.golang.org/grpc: Expose ConnectivityState of a ClientConn (https://github.com/grpc/grpc-go/commit/a5d184a)
- github.com/lightstep/lightstep-tracer-go: Reliable Flushing (https://https://github.com/lightstep/lightstep-tracer-go/commit/e305f42)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17490)
<!-- Reviewable:end -->

irfansharif commented:
See #17495, will need to rebase all the things.

tamird commented:
Ack, done.

petermattis commented:
On principal, let's hold off on merging this until the 1.1 branch is created.

dt commented on Gopkg.toml:
```diff
@@ -20,35 +20,116 @@ required = [
   branch = "master"
 
+# https://github.com/golang/dep/commit/cbc263f introduces a dependency on
+# github.com/go-yaml/yaml which is not compatible with our dependency on
```
ugh. we should consider a .bin/vendor or something, so our `vendor` dir only has to satisfy our actual deps, and not all the transitive deps of all our dev tools too.

tamird replied:
Yes, that's possibly a good idea, but it's a band-aid in this case (this case is the one in which dep is garbage and can't reconcile these import paths).

tamird committed:
Update native deps

tamird committed:
Allow merging Gopkg.toml

I believe this was originally disallowed because we were using
Gopkg.toml as Gopkg.lock.

tamird commented:
After a series of battles with `dep` to get it to work on my spinny-disk laptop, this is rebased and green.

dt commented:
1.1 branch is now cut and this isn't really a sweeping refactor / chaging lots of APIs (that would thus mess with cherrypicks), so I think this is probably okay to merge now, but i'll defer to @bdarnell for a final call.

bdarnell commented:
Touching 86 files is a pretty sizeable PR. And even if it doesn't cause merge conflicts, it still causes build cache invalidations and slows people down on branch switches. I still think that the right time in the cycle to do general dependency updates is after x.y.1 is released.

tamird committed:
ui: appease yarn warnings

tamird commented:
Rebased, now that v1.1.0 is cut. Note that Github's count of modified files is misleading, since the majority are generated files.

```
$ gd master --stat -w --patience -- ':!*.pb.*' ':!*/embedded.go'
 .gitattributes                                   |  12 ++---
 Gopkg.lock                                       | 329 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------
 Gopkg.toml                                       |  87 ++++++++++++++++++++++++++-------
 build/protobuf.mk                                |  10 ++--
 build/style_test.go                              |  29 +++++++++--
 c-deps/protobuf                                  |   2 +-
 c-deps/snappy                                    |   2 +-
 pkg/acceptance/cli_test.go                       |   2 +-
 pkg/acceptance/cluster/docker.go                 |  40 +++++++++------
 pkg/acceptance/cluster/dockercluster.go          |  25 +++++-----
 pkg/acceptance/terrafarm/ssh.go                  |   1 +
 pkg/ccl/sqlccl/backup.go                         |   6 +--
 pkg/ccl/storageccl/export_storage.go             |  28 +++++------
 pkg/cli/cli_test.go                              |   1 +
 pkg/cli/zone.go                                  |  13 +++--
 pkg/cmd/docgen/funcs.go                          |  14 +++---
 pkg/cmd/zerosum/main.go                          |   2 +-
 pkg/config/config_test.go                        |   7 ++-
 pkg/kv/txn_coord_sender_test.go                  |   6 +--
 pkg/kv/txn_test.go                               |   5 +-
 pkg/security/x509.go                             |   2 -
 pkg/server/node.go                               |   5 +-
 pkg/server/status_test.go                        |  28 ++++++-----
 pkg/sql/create.go                                |   8 +--
 pkg/sql/drop.go                                  |  12 ++---
 pkg/sql/ir/base/sexpr.tmpl.go                    |   8 ++-
 pkg/sql/ir/example/base/sexpr.ir.go              | Bin 2166 -> 2693 bytes
 pkg/sql/ir/irgen/.gitignore                      |   1 +
 pkg/sql/logictest/parallel_test.go               |   3 +-
 pkg/sql/logictest/testdata/logic_test/show_trace |  12 ++---
 pkg/sql/monotonic_insert_test.go                 |   7 ++-
 pkg/sql/parser/aggregate_builtins.go             |   5 +-
 pkg/sql/rename.go                                |   5 +-
 pkg/sql/results_writer.go                        |   3 +-
 pkg/sql/txn.go                                   |   5 +-
 pkg/storage/engine/gc.go                         |   3 +-
 pkg/ui/Makefile                                  |  12 ++---
 vendor                                           |   2 +-
 38 files changed, 447 insertions(+), 295 deletions(-)
```

madelynnblue reviewed (approved):
LGTM :shipit:

tamird committed:
github-pull-request-make: print all unexpected diffs

Previously we'd exit on the first diff detected.

tamird committed:
github-pull-request-make: verbose dep ensure

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 58 files at r2, 1 of 1 files at r4, 1 of 1 files at r5, 18 of 18 files at r6, 86 of 86 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17490#-:-KwbNnLC5RESLaKq_LmQ:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
BTW, @tamird just let me know that this also fixes a regular docker flakiness (https://github.com/moby/moby/pull/35144) by bumping the dep. Should triage the Docker flakes that have

> [20:43:47]	[Test Output]
E171017 20:43:47.622111 79 acceptance/cluster/docker.go:326  json: cannot unmarshal string into Go struct field ContainerJSON.Platform of type v1.Platform
panic: runtime error: invalid memory address or nil pointer dereference

in them once this merges.

tamird committed:
storage: tie intent resolution to stopper

tamird committed:
github-pull-request-make: print command directories

tamird commented:
Note that this now pins grpc < 1.7.0 due to https://github.com/grpc/grpc-go/pull/1506#issuecomment-337652759.

tamird committed:
cli: correctly pipe stderr to grep

mjibson commented:
:tada:
